src/Services/ServiceAccountJwtRetriever.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace KeycloakAuthGuard\Services;
+
+use Illuminate\Http\Client\RequestException;
+use Illuminate\Support\Facades\Config;
+use Illuminate\Support\Facades\Http;
+
+readonly class ServiceAccountJwtRetriever implements ServiceAccountJwtRetrieverInterface
+{
+    private string $keycloakBaseUrl;
+
+    private string $realm;
+
+    public function __construct(private string $clientId, private string $clientSecret)
+    {
+        $this->keycloakBaseUrl = trim(Config::get('keycloak.base_url'), '/');
+        $this->realm = Config::get('keycloak.realm');
+    }
+
+    /**
+     * @throws RequestException
+     */
+    public function getJwt(): string
```

Comment:
**[non-blocker]** This is totally subjective, so feel free to ignore if you disagree.

```suggestion
    public function obtainAccessToken(): string
```